cachi2/core/package_managers/bundler/parser.py

```diff
@@ -38,19 +40,32 @@ class _GemMetadata(pydantic.BaseModel):
     name: str
     version: str
 
+    def download_to(self, fs_location: Path) -> None:
+        return None
+
 
 class GemDependency(_GemMetadata):
     """
     Represents a gem dependency.
 
     Attributes:
-        source:     The source URL of the gem.
+        source:     The source URL of the gem as stated in 'remote' field from Gemfile.lock.
         checksum:   The checksum of the gem.
     """
 
     source: str
     checksum: Optional[str] = None
 
+    @cached_property
+    def remote_location(self) -> str:
+        """Return remote location to download this gem from."""
+        return f"{self.source}/gems/{self.name}-{self.version}.gem"
+
+    def download_to(self, fs_location: Path) -> None:
```

Comment:
I believe we default to using `RootedPath` to make sure that all filepaths will not resolve outside of the repository root.

Reply:
Good point, thank you.